src/core.c/Main.pm6

```diff
@@ -288,7 +288,7 @@ my sub RUN-MAIN(&main, $mainline, :$in-as-argsfiles) {
     }
 
     sub find-candidates($capture) {
-        &main.^name eq 'Sub'
+        &main.^name ~~ /^ Sub [ '+{Callable[' .*? ']}' ]? $/
```

Comment:
I wonder why we test for a class name here instead of doing a normal type check? The latter wouldn't be affected by mixins.

Reply:
I did throw this together quite quickly last night, I'll take a closer look later today to see how this can be improved.